ui/common/src/main/java/com/example/ui/common/BaseViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.ui.common
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+
+/**
+ * @author yaya (@yahyalmh)
+ * @since 29th September 2022
+ */
+
+abstract class BaseViewModel<T : UIState, E : UIEvent>(initialState: T) : ViewModel() {
+    private var internalSate: MutableState<T> = mutableStateOf(initialState)
```

Comment:
could be val - you are creating one object of type MutableState<T> during initialisation. Later on setState method just change underlying value in object - it's still the same MutableState but with different value